core/iwasm/common/gc/gc_common.c

```diff
@@ -42,37 +41,48 @@ wasm_defined_type_is_struct_type(WASMType *const def_type)
 bool
 wasm_defined_type_is_array_type(WASMType *const def_type)
 {
-    /* TODO */
-    return false;
+    return wasm_type_is_array_type(def_type);
 }
 
 uint32
 wasm_func_type_get_param_count(WASMFuncType *const func_type)
 {
-    /* TODO */
-    return 0;
+    return func_type->param_count;
 }
 
 wasm_ref_type_t
 wasm_func_type_get_param_type(WASMFuncType *const func_type, uint32 param_idx)
 {
-    wasm_ref_type_t ref_type = { 0 };
-    /* TODO */
+    bh_assert(param_idx < func_type->param_count);
+    WASMRefTypeMap ref_type_maps = func_type->ref_type_maps[param_idx];
```

Comment:
Should firstly get value type from `func_type->types[param_idx]`
And then use `wasm_is_type_multi_byte_type` to check if it's a ref type with heap type, then traverse `func_type->ref_type_maps` to get the map with same `index` with `param_idx`, then get its `ref_type`

You can search `wasm_is_type_multi_byte_type` to check the usage

Reply:
get~ thanks~